chore: update release doc, push git tag before uploading artifacts (#2747)
apache/iceberg-rust: website/src/release.md

The script checks license headers against the generated source archive, not the live Git worktree. If enabled, SVN upload runs after local artifact verification and before RC tag creation. The script creates the signed RC tag as the final release step, then prints a draft VOTE email for `dev@iceberg.apache.org`.

> By default the script does not upload to SVN (`--upload_svn 0`), so the source artifacts stay local at this stage. Push the RC tag to GitHub first, let the publish workflow build and publish the RC wheels, and only then upload the source artifacts to ASF dev dist.

Reviewer: nit, I think it's clearer if we just say that it stays local and we'll handle upload later.

```suggestion
By default the script does not upload to SVN (`--upload_svn 0`), so the source artifacts stay local at this stage.
The artifacts will be uploaded in a later step.
```

### Push the release candidate tag

The script does not push the RC tag. Review the output, then push the tag manually:

```shell
git push origin "v${iceberg_version}-rc.${rc}"
```

> Pushing a `v${iceberg_version}-rc.${rc}` tag triggers the [publish workflow](https://github.com/apache/iceberg-rust/blob/main/.github/workflows/publish.yml).
> For a release candidate tag, crate publishing is skipped (it only runs for non-pre-release tags) and the workflow builds and publishes the pyiceberg-core RC wheels to PyPI as a PEP 440 pre-release (for example `0.9.1rc2`).

Reviewer: I think we can link directly to the actions page?

```suggestion
Pushing a `v${iceberg_version}-rc.${rc}` tag triggers the [publish workflow](https://github.com/apache/iceberg-rust/actions/workflows/publish.yml).
For a release candidate tag, crate publishing is skipped (it only runs for non-pre-release tags) and the workflow builds and publishes the pyiceberg-core RC wheels to PyPI as a PEP 440 pre-release (for example `0.9.1rc2`).
```

(why do we not publish RCs on crates.io anyway?)

svn commit -m "Prepare Apache Iceberg Rust ${iceberg_version} RC${rc}"
```

> Alternatively, `dev/release/create_rc.sh` can upload the artifacts for you with `--upload_svn 1`. Use this only once you are confident the RC is good, since it uploads before the tag is pushed and the publish workflow runs.

Reviewer: aside: should we update the script to block committing to SVN before the Git tag is pushed?
